config/libc.in

```diff
@@ -52,6 +56,7 @@ choice
     prompt "Threading implementation to use:"
     default THREADS_NATIVE         if LIBC_SUPPORT_THREADS_NATIVE
     default THREADS_THREADS_LT     if LIBC_SUPPORT_THREADS_LT && ! LIBC_SUPPORT_THREADS_NATIVE
+    default THREADS_THREADS_POSIX  if LIBC_SUPPORT_THREADS_POSIX && ! LIBC_SUPPORT_THREADS_NATIVE && ! LIBC_SUPPORT_THREADS_LT
```

Comment:
These lines become too long, and they are not needed: Kconfig selects the first `default` statement with matching if-condition. So this can be rewritten as:
````
default THREADS_NATIVE         if LIBC_SUPPORT_THREADS_NATIVE
default THREADS_THREADS_LT     if LIBC_SUPPORT_THREADS_LT
default THREADS_THREADS_POSIX  if LIBC_SUPPORT_THREADS_POSIX
default THREADS_NONE
````
And, we can drop `LIBC_SUPPORT_THREADS_ANY` completely.